### DISCUSSION TITLE
Properly implemeting TraCI/TraCICommandInterface

### DISCUSSION THREAD

**Discussion by HObaid00** [riebl/artery#354] [Q&A] [answered]
Hi, I get confused when trying to get an instance for TraCI and if it is wise to implement a new instance or how to get the currently running one for the Vehicle. I would like to know the requirements and how to initialize them for the TraCI API. If anyone has experience in using it please do let me know. 
Best Regards, and thank you in advance.

**Comment by riebl** [comment]
The `TraCICommandInterface` is part of Veins, but we don't use it in Artery. Instead, we bundle the full SUMO TraCI C++ API. One way to gain access to the API object is the `traci::Core::getAPI` method of the global _World.traci.core_ module.

**Reply by HObaid00** [reply]
Hi Raphael, thank you for your response. I followed up on that when we got the World.traci.core module through the getAPI is this a global instance or will this then be a new instance? Also, what are the necessary parts to get for a vehicle controller instance of the current vehicle you are working with?

**Reply by riebl** [reply, marked as answer]
There is only a single API instance shared among all modules. This API object is a sub-class of the official SUMO `TraCIAPI` class, i.e. everything supported by SUMO is available. Vehicles are identified by unique IDs in SUMO and these string parameters are usually called `vehicleID` in the SUMO API. You can get the very same ID from `VehicleController::getVehicleId`.
